### ISSUE TITLE
[Usage/Bug] Quantizing BatchNorm

### ISSUE THREAD

**Issue by skoppula** [ppwwyyxx/tensorpack#443]
Hi there!

I was modifying the DoReFa net, and was interested to know how you guys (if at all) quantized BatchNorm. The script in the repo does not quantize the normalization layers. 

I understand Megvii had an FPGA demo, so some level of quantization occurred somewhere.

Any thoughts? It doesn't seem like the script rolls the BatchNorm into the FC/Conv layers either.

As a side note, I tried including the BatchNorm moving average variables as part of the quantized variables, and got some wild TF bugs (e.g. the moving averages is a tensor after wrapping with quantization layer, so assign_sub() in tf.Variable  doesn't work (necessary for moving average).)

**Comment by ppwwyyxx** [comment]
During training there is no need for quantization because batchnorm is fast.
During inference, quantized batch norm becomes `quantize([quantized number] x [constant] + [constant])`, which can be implemented by a lookup table on FPGA.

This is not a tensorpack question. I suggest you email the authors about the paper details.
